vault/provider.go

```diff
@@ -108,18 +108,36 @@ func (p *VaultProvider) Retrieve() (credentials.Value, error) {
 		}
 
 		// session lookup missed, we need to create a new one
-		creds, err := p.getMasterCreds()
-		if err != nil {
-			return credentials.Value{}, err
-		}
-
-		session, err = p.getSessionToken(&creds)
-		if err != nil {
-			return credentials.Value{}, err
-		}
+		if profile, exists := p.config.Profile(p.profile); exists && profile.SourceProfile != "" {
```

Comment:
Reading this code without the PR context takes a while to parse intent. Perhaps update the comment to make it clear what is happening and why?

Reply:
Updated the comment to explain the control flow.